src/app/game-state/log.service.ts

```diff
@@ -35,7 +36,13 @@ export class LogService {
   followerLog: Log[] = [];
   currentLog: Log[] = [];
 
-  constructor() {
+  constructor(
+    
+    mainLoopService: MainLoopService
+  ) {
+    mainLoopService.frameSubject.subscribe(() => {
+      this.updateLogTopics();
```

Comment:
This seems like it does a lot of extra sorting. How's the performance compare? And could we optimize further by adding a flag that gets set in addLogMessage so we know if there's new stuff to sort or if we can skip it this tick?

Reply:
This isn't a tick, it's a 25ms, so hundreds of ticks could go by. Let's not forget that when the tickSubject triggers it loops its next method. And yeah, this improves a lot, about as much as the BigNumber pipe fix. I'd love to be able to do this update once per frame, preferably pre-draw, but I don't know what to listen to. As is, it's roughly twice a frame.